### DISCUSSION TITLE
WT32-SC01 Plus con ESP32-S3 e Display Touch 3.5"

### DISCUSSION THREAD

**Discussion by AprileNunzio** [HASwitchPlate/openHASP#610] [Q&A]
I have WT32-SC01 Plus with ESP32-S3 and 3.5" Touch Display, I managed to install OpenHasp, I understood how to install the example theme in pages.jsonl, but I can't understand how to make an LED light up if I click on a button.
Could someone kindly help me?

**Comment by hb020** [comment]
3 options:
* the preferred way: control the led through a remote machine that listens to the MQTT messages sent by the SC01Plus
* a bit more complicated but self contained: if the led is directly connected to the SC01Plus (via the I/O connector), compile in custom code. Search for HASP_USE_CUSTOM.
* totally distorted way: via a web server that controls the led. You'd need to create a page in the SC01Plus that has a button and an image control. Use the button to tell the image control to request the image from a web server. An image can fetch images from local storage (L:/image.png), or from a web server (http://webserver/image.png). Use the latter. Then, that web server should return some image, and light up the led.

**Comment by AprileNunzio** [comment]
Isn't it possible to have a demonstration video that explains everything?

Il lun 22 gen 2024, 11:56 hb020 ***@***.***> ha scritto:

**Comment by hb020** [comment]
I can't provide a video, but I could help you a bit (outside of this forum) if you tell me what tools and skills you have. There are minima however.

For the 3 options I mentioned:
* option 1 (MQTT), as a minimum you need to know what MQTT is, and have one of the integrations mentioned in https://openhasp.haswitchplate.com/0.7.0/integrations/home-assistant/howto/, or have an installation of mosquitto for example.
* option 2 (compile), as a minimum you need to know to program in C, and be at ease with making electronics circuits
* option 3 (web server), as a minimum you need sysadmin skills plus know some web server language, like node ou python.

You can also abandon openHASP and program the SC01Plus directly, via the "arduino" methods. There are several repositories out there that have examples. But as a minimum you'd need the skill level of option 2.
